packages/app/components/ManageTabContent.tsx

```diff
@@ -90,10 +92,47 @@ const ManageTabContent = () => {
                 from: address,
             })
             .then((r: string[]) => {setMAddresses(r)})
-            .catch(console.error)
+            .then(() => {
+                // Get details of all private memberships to populate dropdown
+                if(mAddresses.length > 0) {
+                    setShowLoading(true);
+                    const promises = mAddresses.map((mAddr) => {
+                        const contract = new web3.eth.Contract(getAbiFromJson(passportJson));
+                        contract.options.address = mAddr;
+                        return (contract.methods.get() as ContractSendMethod).call().then((p) => {
+                            if (p[7]) {
+                                const data: MembershipDetailMap = {};
+                                data[mAddr] = {
+                                    name: p[0],
+                                    symbol: p[1],
+                                    supply: p[2],
+                                    price: web3.utils.fromWei(p[3], "ether")
+                                };
+                                return data;
+                            }
+                            else {
+                                return {};
+                            }
+                        });
+                    });
+                    Promise.all(promises).then((values) => {
```

Comment:
nit: we could just `return Promise.all(...)` here, making the `catch`/`finally` on lines 122-126 redundant as it will be caught by line 132